src/Hng.Application/Features/Categories/Handlers/Commands/CreateCategoryCommandHandler.cs

```diff
@@ -0,0 +1,54 @@
+﻿using AutoMapper;
+using Hng.Application.Features.Categories.Commands;
+using Hng.Application.Features.Categories.Dtos;
+using Hng.Application.Shared.Dtos;
+using Hng.Domain.Entities;
+using Hng.Infrastructure.Repository.Interface;
+using MediatR;
+
+namespace Hng.Application.Features.Categories.Handlers.Commands
+{
+    public class CreateCategoryCommandHandler : IRequestHandler<CreateCategoryCommand, SuccessResponseDto<CategoryDto>>
+    {
+        private readonly IRepository<Category> _categoryRepository;
+        private readonly IMapper _mapper;
+
+        public CreateCategoryCommandHandler(IRepository<Category> categoryRepository, IMapper mapper)
+        {
+            _categoryRepository = categoryRepository;
+            _mapper = mapper;
+        }
+
+        public async Task<SuccessResponseDto<CategoryDto>> Handle(CreateCategoryCommand request, CancellationToken cancellationToken)
+        {
+            if (string.IsNullOrWhiteSpace(request.Name))
+            {
+                return new SuccessResponseDto<CategoryDto>
+                {
+                    Data = null,
+                    Message = "Category name is required.",
+                    StatusCode = 400
+                };
+            }
+
```

Comment:
Add a check for if the category name already exists, then it should give a bad request category already exists